waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,3 +337,48 @@
                     :interval 15
                     :timeout timeout-secs)))))
         (log/warn "skipping test as INTEGRATION_TEST_BAD_IMAGE is not specified")))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (if-let [{:strs [flag offset]} (-> waiter-url
+                                         get-k8s-proxy-options
+                                         :reverse-proxy-parameters)]
+        (do
+          (let [{:keys [service-id] :as response}
+                (make-request-with-debug-info
+                  {:x-waiter-name (rand-name)
+                   (keyword (str "x-waiter-env-" flag)) "yes"}
+                  #(make-kitchen-request waiter-url % :method :get :path "/request-info"))]
+            (with-service-cleanup
+              service-id
+              (assert-response-status response http-200-ok)
+              (testing "Expected envoy specific headers are present in both request and response"
+                (let [response-body (try-parse-json (:body response))
+                      response-headers (:headers response)]
+                  (is (contains? (get response-body "headers") "x-envoy-expected-rq-timeout-ms"))
+                  (is (contains? response-headers "x-envoy-upstream-service-time"))
+                  (is (= "envoy" (get response-headers "server")))))))
+          (let [{:keys [service-id] :as response}
+                (make-request-with-debug-info
+                  {:x-waiter-name (rand-name)
+                   (keyword (str "x-waiter-env-" flag)) "yes"}
+                  #(make-kitchen-request waiter-url % :method :get :path "/environment"))]
+            (with-service-cleanup
```

Comment:
Instead of creating+deleting this same service twice, we should reuse the service across both requests.

Reply:
Resolved